fbw-a32nx/src/wasm/fadec_a32nx/src/Fadec/EngineControlA32NX.cpp

```diff
@@ -1109,103 +1108,69 @@ void EngineControl_A32NX::updateThrustLimits(double                  simulationT
 #endif
 }
 
-/*
- * Previous code - call to it was already commented out and this function was not in use.
- * Keeping it to make completing/fixing it easier.
- * It is not migrated to the cpp framework yet.
- *
- * /// <summary>
+/// <summary>
 /// FBW Oil Qty, Pressure and Temperature (in Quarts, PSI and degree Celsius)
 /// Updates Oil with realistic values visualized in the SD
 /// </summary>
-void updateOil(int engine, double imbalance, double thrust, double simN2, double deltaN2, double deltaTime, double ambientTemp) {
-  double steadyTemperature;
-  double thermalEnergy;
-  double oilTemperaturePre;
-  double oilQtyActual;
-  double oilTotalActual;
-  double oilQtyObjective;
-  double oilBurn;
-  double oilIdleRandom;
-  double oilPressure;
-
-//--------------------------------------------
-// Engine Reading
-//--------------------------------------------
-if (engine == 1) {
-steadyTemperature = simVars->getEngine1EGT();
-thermalEnergy = thermalEnergy1;
-oilTemperaturePre = oilTemperatureLeftPre;
-oilQtyActual = simVars->getEngine1Oil();
-oilTotalActual = simVars->getEngine1OilTotal();
-} else {
-steadyTemperature = simVars->getEngine2EGT();
-thermalEnergy = thermalEnergy2;
-oilTemperaturePre = oilTemperatureRightPre;
-oilQtyActual = simVars->getEngine2Oil();
-oilTotalActual = simVars->getEngine2OilTotal();
-}
+void EngineControl_A32NX::updateOil(int         engineIdx,
+                                    EngineState engineState,
+                                    double      imbalance,
+                                    double      thrust,
+                                    double      simN2,
+                                    double      deltaTime,
+                                    bool        isOnGround,
+                                    double      ambientTemp) {
+  const int engineImbalanced = imbalanceExtractor(imbalance, 1);
 
-//--------------------------------------------
-// Oil Temperature
-//--------------------------------------------
-if (simOnGround == 1 && engineState == 0 && ambientTemp > oilTemperaturePre - 10) {
-oilTemperature = ambientTemp;
-} else {
-if (steadyTemperature > oilTemperatureMax) {
-  steadyTemperature = oilTemperatureMax;
-}
-thermalEnergy = (0.995 * thermalEnergy) + (deltaN2 / deltaTime);
-oilTemperature = poly->oilTemperature(thermalEnergy, oilTemperaturePre, steadyTemperature, deltaTime);
-}
+  //--------------------------------------------
+  // Oil Temperature
+  //--------------------------------------------
+  const double egt               = simData.engineEgt[engineIdx]->get();
+  const double oilTemperaturePre = simData.engineOilTemperature[engineIdx]->get();
+  const double oilTemperatureMax = MAX_OIL_TEMP_NOMINAL + (engineIdx + 1) == engineImbalanced ? imbalanceExtractor(imbalance, 8) : 0;
+  double       oilTemperature;
+  if (isOnGround && engineState == EngineState::OFF && ambientTemp > oilTemperaturePre - 10) {
```

Comment:
This also doesn't really make sense to me. It shouldn't really matter whether the plane is on the ground or not?

Reply:
I wasn't really looking to work on the code any further than needed, only re-enable it. I think this does make some sense as a sanity check for spawn situations where the Lvar might have an old value.